ignite/services/plugin/cache_test.go

```diff
@@ -0,0 +1,76 @@
+package plugin_test
+
+import (
+	"net"
+	"testing"
+
+	hplugin "github.com/hashicorp/go-plugin"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ignite/cli/ignite/services/plugin"
+)
+
+func TestPluginCacheAdd(t *testing.T) {
+	const path = "/path/to/awesome/plugin"
+	unixFD, _ := net.ResolveUnixAddr("unix", "/var/folders/5k/sv4bxrs102n_6rr7430jc7j80000gn/T/plugin193424090")
+
+	rc := hplugin.ReattachConfig{
+		Protocol:        hplugin.ProtocolNetRPC,
+		ProtocolVersion: hplugin.CoreProtocolVersion,
+		Addr:            unixFD,
+		Pid:             24464,
+	}
+
+	err := plugin.WritePluginConfigCache(path, rc)
+	require.NoError(t, err)
+
+	c := hplugin.ReattachConfig{}
+	err = plugin.ReadPluginConfigCache(path, &c)
+	require.NoError(t, err)
+}
+
+func TestPluginCacheDelete(t *testing.T) {
+	const path = "/path/to/awesome/plugin"
+	unixFD, _ := net.ResolveUnixAddr("unix", "/var/folders/5k/sv4bxrs102n_6rr7430jc7j80000gn/T/plugin193424090")
+
+	rc := hplugin.ReattachConfig{
+		Protocol:        hplugin.ProtocolNetRPC,
+		ProtocolVersion: hplugin.CoreProtocolVersion,
+		Addr:            unixFD,
+		Pid:             24464,
+	}
+
+	err := plugin.WritePluginConfigCache(path, rc)
+	require.NoError(t, err)
+
+	err = plugin.DeletePluginConfCache(path)
+	require.NoError(t, err)
+
+	c := hplugin.ReattachConfig{}
+	// there should be an error after deleting the config from the cache
+	err = plugin.ReadPluginConfigCache(path, &c)
+	require.Error(t, err)
```

Comment:
Can we test the failing cases of these functions? Looks like we are only testing passing cases.

Reply:
good point, added further test cases